dev-client/__tests__/integration/OfflineSnackbar.test.tsx

```diff
@@ -0,0 +1,320 @@
+          <ProjectViewScreen projectId="1" />
+        </Portal.Host>
+      </PaperProvider>,
+    );
+
+    expect(screen.queryByTestId(snackbarTestId)).toBeOnTheScreen();
+  });
+
+  // TODO-cknipe: 
+  // Issue: I expect the snackbar to be on the screen after the first render (this is passing), 
+  // but not to be on the screen at the end (this is failing).
+  // - Putting a console.log in the snackbar component shows the component's visible prop is false on the
+  //   last render before the test's expect statement.
+  // - I hypothesized maybe this is because the snackbar is still in React's virtual DOM, but not visible. 
+  //   - In tests it seems like if the snackbar doesn't start on the screen, it's (correctly) not detected 
+  //     by toBeOnTheScreen(). But once it has been put on the screen, I have not found a way to stop it
+  //     being detected, even when I try to remove it. This behavior seems incongruent with the behavior of
+  //     both the Components hierarchy, and of what the user sees.
+  //   - When I look in the product with React Native DevTools, if the snackbar is visible to the user, 
+  //     there is a Snackbar in the Components hierarchy with visible: true. 
+  //     If it is not visible, there is a Snackbar in the Components hierarchy with visible: false.
+  //   - The presence of the Portal seems irrelevant, at least removing the Portal.Host has no
+  //     effect on the outcome of the tests.
+  test('disappears when returning online', async () => {
+    console.log('------- TEST: Disappears when returning online -------');
+    useIsOfflineMock.mockReturnValue(true);
+
+    const screen = render(
+      <PaperProvider>
+        <Portal.Host>
+          <OfflineSnackbar />
+          <ProjectViewScreen projectId="1" />
+        </Portal.Host>
+      </PaperProvider>,
+      {
+        route: 'PROJECT_VIEW',
+        initialState: testState,
+      },
+    );
+
+    expect(screen.queryByTestId(snackbarTestId)).toBeTruthy();
+
+    // This fails, and doesn't seem like an ideal way to test this anyway
+    // expect(screen.queryByTestId(snackbarTestId).props.visible).toBe(true);
+
+    console.log('Starting 2nd render');
+    useIsOfflineMock.mockReturnValue(false);
+    screen.rerender(
+      <PaperProvider>
+        <Portal.Host>
+          <OfflineSnackbar />
+          <ProjectViewScreen projectId="1" />
+        </Portal.Host>
+      </PaperProvider>,
+    );
+
+    await waitForElementToBeRemoved(screen.queryByTestId(snackbarTestId))
+      .then(() => {
+        console.log('Yup'); // This does not happen
+      })
+      .catch(() => {
+        console.log('Nope'); // This happens
+      });
+
+    // These all fail
+    expect(screen.queryByText(snackbarText)).toBeFalsy();
+    expect(screen.queryByText(snackbarText)).not.toBeOnTheScreen();
+    expect(screen.queryByTestId(snackbarTestId)).toBeFalsy();
+    expect(screen.queryByTestId(snackbarTestId)).not.toBeOnTheScreen();
```

Comment:
So, in order to make it work you should do something like this
```suggestion
  // Wait for the animation to complete
  act(() => {
    jest.advanceTimersByTime(1500);
  });

  // These assertions should now pass
  expect(screen.queryByText(snackbarText)).toBeFalsy();
  expect(screen.queryByText(snackbarText)).not.toBeOnTheScreen();
  expect(screen.queryByTestId(snackbarTestId)).toBeFalsy();
  expect(screen.queryByTestId(snackbarTestId)).not.toBeOnTheScreen();
```